### PR TITLE
[PLT-7829] Add resetCreatePost to make PostDeletedModal to show always after replying to a deleted post

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add resetCreatePost to make PostDeletedModal to show always after replying to a deleted post

#### Ticket Link
Jira ticket: [PLT-7829](https://mattermost.atlassian.net/browse/PLT-7829)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

#### Test Information
This PR was tested on: Chrome/FF on Mac

### CODE REVIEW

File: src/action_types/posts.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ export default keyMirror({
     CREATE_POST_REQUEST: null,
     CREATE_POST_SUCCESS: null,
     CREATE_POST_FAILURE: null,
+    CREATE_POST_RESET: null,
```
Comment (enahum) [inline src/action_types/posts.js]: can we name this `CREATE_POST_RESET_REQUEST`

File: src/actions/posts.js [hunk 2]
```diff
@@ -122,6 +122,14 @@ export function createPost(post, files = []) {
     };
 }
 
+export function resetCreatePost() {
```
Comment (enahum) [inline src/actions/posts.js]: also can we rename the action to be `resetCreatePostRequest`

in this type of action you don't really need a thunk so you can do:

```js
export function resetCreatePostRequest() {
    return {type: PostTypes.CREATE_POST_RESET_REQUEST};
}
```

**Comment by enahum** [comment]
@saturninoabril looks like the tests are failing for this one

**Comment by saturninoabril** [comment]
I'll check this out, not sure why it's failing here `Actions.Posts retry failed post`.

**Pull request merged**